src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -588,12 +583,12 @@ static void grpc_cancel_ares_request_locked_impl(grpc_ares_request* r) {
 void (*grpc_cancel_ares_request_locked)(grpc_ares_request* r) =
     grpc_cancel_ares_request_locked_impl;
 
+// ares_library_init and ares_library_cleanup are currently no-op except under
+// Windows. Calling them may cause race conditions when other parts of the
+// binary calls these functions concurrently.
+#ifdef GPR_WINDOWS
 grpc_error* grpc_ares_init(void) {
-  gpr_once_init(&g_basic_init, do_basic_init);
-  gpr_mu_lock(&g_init_mu);
```

Comment:
Why isn't the mutex needed anymore?

Reply:
Because this is called through a plugin (dns_resolver) and thus happen under grpc_init/grpc_shutdown's global lock.